src/utils/castFilter.ts

```diff
@@ -17,7 +17,47 @@ export default function castFilter<T extends ADPBaseModel>(
       obj[key] = path(key).cast(obj[key], null, null);
     } catch (e) {}
 
-    if (allowedRegexes.includes(key) && typeof obj[key] === "string") {
+    /**  Parse MongoDB Query Operators **/
+    let splittedKey = key.split("_");
+    if (
+      splittedKey.length > 1 &&
+      ["eq", "gt", "gte", "in", "lt", "lte", "ne", "nin"].includes(
+        splittedKey[splittedKey.length - 1]
+      )
+    ) {
+      let operator = splittedKey.pop();
```

Comment:
This is somewhat hard to read IMO. Could add comments to make it easier to enhance readability.